core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -557,7 +557,7 @@ private[spark] class TaskSetManager(
         lastLaunchTime += localityWaits(currentLocalityIndex)
         currentLocalityIndex += 1
         logDebug(s"Moving to ${myLocalityLevels(currentLocalityIndex)} after waiting for " +
-          s"${localityWaits(currentLocalityIndex)}ms")
+          s"${localityWaits(currentLocalityIndex - 1)}ms")
```

Comment:
Pull this out into a `val` above, use it above, and then just reuse it here. It's clearer